flake.nix

```diff
@@ -0,0 +1,24 @@
+{
+    description = "Stregsystemet-rs nix flake";
+    inputs = {
+        nixpkgs.url = "nixpkgs/nixos-24.05";
+    };
+    outputs = { self, nixpkgs }: let 
+        system = "x86_64-linux";
+        pkgs = import nixpkgs {inherit system;};
+    in {
+        devShells.${system}.default = pkgs.mkShellNoCC {
+            DATABASE_URL = "postgres://stregsystemet:password@localhost/stregsystemet";
+            packages = with pkgs; [
+                cargo
+                rustc
+                postgresql
+            ];
+        };
+        packages.${system}.default = pkgs.rustPlatform.buildRustPackage {
+            name = "stregsystemet-rs";
+            cargoSha256 = "sha256-VY0UXted3xbpoHTscFZ1gc/n8xHPU4JYjJeDpzUkC+4=";
```

Comment:
Do we need to keep this updated? If so we should have a github actions bot update this for us.

Reply:
yes, its the hash of the cargo lock for dependencies

the alternative would be to list **all** dependencies from nixpkgs

you could probably move the package to its own file and generate it, but it'll be very jank